tests/api2/test_pool_dataset_unlock.py

```diff
@@ -0,0 +1,159 @@
+import contextlib
+
+import pytest
+
+from auto_config import pool_name
+from middlewared.test.integration.assets.account import user
+from middlewared.test.integration.utils import call, ssh
+from protocols import SMB
+from samba import ntstatus, NTSTATUSError
+
+
+SMB_PASSWORD = 'Abcd1234'
+SMB_USER = 'smbuser999'
+
+
+def passphrase_encryption():
+    return {
+        'encryption_options': {
+            'generate_key': False,
+            'pbkdf2iters': 100000,
+            'algorithm': 'AES-128-CCM',
+            'passphrase': 'passphrase',
+        },
+        'encryption': True,
+        'inherit_encryption': False,
+    }
+
+
+@contextlib.contextmanager
+def dataset(name, options=None):
+    assert '/' not in name
+
+    dataset = f'{pool_name}/{name}'
+    call('pool.dataset.create', {'name': dataset, **(options or {})})
+    call('filesystem.setperm', {'path': f'/mnt/{dataset}', 'mode': '777'}, job=True)
+
+    try:
+        yield dataset
+    finally:
+        assert call('pool.dataset.delete', dataset)
+
+
+@contextlib.contextmanager
+def smb_share(name, path, options=None):
```

Comment:
The standard asset for samba share also exists

Reply:
Thanks for pointing these out